johnzon-core/src/main/java/org/apache/johnzon/core/JsonGeneratorImpl.java

```diff
@@ -449,8 +449,8 @@ public void close() {
         try {
             if (ex == null) {
                 flushBuffer();
+                writer.close();
```

Comment:
That one is an intended change to pass [jsonGeneratorStreamNotClosedTest](https://github.com/jakartaee/jsonp-api/blob/2.1.1-tck/tck/tck-tests/src/main/java/ee/jakarta/tck/jsonp/api/jsongeneratortests/ClientTests.java#L2094)

Reply:
Hmm, how do we guarantee writer is closed when ex != null? didnt find the call and I don't see us providing a leaking impl cause we had an exception - we can request the spec to fix this bug which was not there before if it is written down if not we just challenge the tck

Reply:
We can't challenge the TCK. The javadoc clearly says it https://jakarta.ee/specifications/jsonp/2.1/apidocs/jakarta.json/jakarta/json/stream/jsongenerator#close()

>  The underlying stream is closed only if complete JSON object is written. In case of incomplete object JsonGenerationException is thrown and underlying stream is not closed.

We can create an issue to fix the specification and therefor fix the TCK accordingly. But as per the current specification version, the challenge would be invalid. I have no idea what was the reason.

Reply:
We guarantee the writer is _not_ closed when an exception is thrown, looks like this has been specified for json-p 2.1 [here](https://github.com/jakartaee/jsonp-api/issues/156)

Also see [javadocs](https://jakarta.ee/specifications/jsonp/2.1/apidocs/jakarta.json/jakarta/json/stream/jsongenerator#close()):
> The underlying stream is closed only if complete JSON object is written. In case of incomplete object JsonGenerationException is thrown and underlying stream is not closed.

Reply:
Here is the explaination https://github.com/jakartaee/jsonp-api/issues/156

Reply:
Ok, then let's add a toggle disabled by default to support that in the tck module, it is not responsible to leak by default IMHO